src/app/tree/checkbox-tree/terra-checkbox-tree.component.ts

```diff
@@ -87,4 +81,85 @@ export class TerraCheckboxTreeComponent extends TerraBaseTreeComponent
             }
         }
     }
+
+    private upRecursiveCheckParentLeafs(leaf:TerraLeafInterface):void
+    {
+        let allChildrenChecked:boolean = true;
+        let noChildChecked:boolean = true;
+        let isIndeterminate:boolean = false;
+
+        if(leaf.leafParent)
```

Comment:
Was wird hier überprüft und warum werden die einzelnen Wert gesetzt?
Als Erklärung: wenn ich nun hier irgendetwas anpassen müsste, könnte ich das nicht so ohne weiteres, sondern müsste erst das if komplett verstehen. Vllt gibt es eine für sich sprechende elegantere Lösung :)
z.b. indem Teile in Funkionen ausgelagert werden.